File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/writecreddef/v0_6/RevocationRegistryConfig.java
package com.evernym.verity.sdk.protocols.writecreddef.v0_6;

import com.evernym.verity.sdk.utils.AsJsonObject;
import org.json.JSONObject;

/**
 * Holder of revocation registry configuration. This class holds a JSON object and is not intended to be created
 * directly. The versioned WriteCredentialDefinition interface provides helper functions to create this holder
 * class.
 *
 * @see WriteCredentialDefinitionV0_6#disabledRegistryConfig()
 */
public class RevocationRegistryConfig implements AsJsonObject {
    private final JSONObject data;

    /**
     * Constructs the object with the given JSON object
     *
     * @param data the given configuration as expressed in JSON object
     */
    public RevocationRegistryConfig(JSONObject data) {
        this.data = data;
    }

    /**
     * Convert this object to a JSON object
     *
     * @return this object as a JSON object
     */
    @Override
    public JSONObject toJson() {
        return data;
    }
}

File: samples/sdk/dotnet-example-app-tmp/VeritySDK/Protocols/WriteCredDef/v0_6/RevocationRegistryConfig.cs
using System;
using System.Json;
using VeritySDK.Utils;

namespace VeritySDK.Protocols.WriteCredDef
{
    /// <summary>
    /// Holder of revocation registry configuration. This class holds a JSON object and is not intended to be created
    /// directly. The versioned WriteCredentialDefinition interface provides helper functions to create this holder class.
    /// </summary>
    public class RevocationRegistryConfig : AsJsonObject
    {
        private JsonObject data;

        /// <summary>
        /// Constructs the object with the given JSON object 
        /// </summary>
        /// <param name="data">the given configuration as expressed in JSON object</param>
        public RevocationRegistryConfig(JsonObject data)
        {
            this.data = data;
        }

        /// <summary>
        /// Convert this object to a JSON object
        /// </summary>
        /// <returns>this object as a JSON object</returns>
        public JsonObject toJson()
        {
            return data;
        }
    }
}